core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1158,12 +1158,18 @@ public int getNumberOfPinnedFiles() {
   public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
       FileDoesNotExistException, DirectoryNotEmptyException, InvalidPathException,
       AccessControlException {
+    List<Inode<?>> deletedInodes;
     Metrics.DELETE_PATHS_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
          LockedInodePath inodePath = mInodeTree.lockFullInodePath(path, InodeTree.LockMode.WRITE)) {
       mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
       mMountTable.checkUnderWritableMountPoint(path);
-      deleteAndJournal(inodePath, options, journalContext);
+      deletedInodes = deleteAndJournal(inodePath, options, journalContext);
```

Comment:
I think the correct paradigm might be to return things via a context, but the context is only about Journal right now. This is still safe, but it may result in some orphan blocks on failures.

Maybe add a TODO to return the list via a context.